registry/remote/repository.go

```diff
@@ -114,6 +128,28 @@ func NewRepository(reference string) (*Repository, error) {
 	}, nil
 }
 
+// SetReferrersCapabilityOnce sets the capability of Referrers API for the
+// repository. true: capable; false: not capable.
+// SetReferrersCapabilityOnce is valid only when it is called for the first time.
+// - When the capability is set to true, the Referrers() function will always
+// request the Referrers API. Reference: https://github.com/opencontainers/distribution-spec/blob/main/spec.md#listing-referrers
+// - When the capability is set to false, the Referrers() function will always
+// request the Referrers Tag. Reference: https://github.com/opencontainers/distribution-spec/blob/main/spec.md#referrers-tag-schema
+// - When the capability is not set, the Referrers() function will automatically
+// determine which API to use.
+func (r *Repository) SetReferrersCapabilityOnce(capable bool) {
```

Comment:
The term `Once` is a bit confusing.
```suggestion
func (r *Repository) SetReferrersCapability(capable bool) {
```